activejob/lib/active_job/queue_adapters/async_adapter.rb

```diff
@@ -110,6 +110,11 @@ def shutdown(wait: true)
         def executor
           immediate ? @immediate_executor : @async_executor
         end
+
+        private
+          def processor_count
+            Integer(Concurrent.available_processor_count || Concurrent.processor_count)
```

Comment:
Why is `Kernel#Integer` necessary?

Reply:
@MatheusRich Cause available processors return float (or nil), we should convert it as integer.